### PR TITLE
fixed ordering for consensus group after flag activation

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
## Reasoning behind the pull request
- This is follow up PR for #6436
  
## Proposed changes
- This PR actually adds the fixed ordering

## Testing procedure
- check description from #6436

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: integrationTests/testProcessorNode.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ import (
 	"github.com/multiversx/mx-chain-go/storage/txcache"
 	"github.com/multiversx/mx-chain-go/testscommon"
 	"github.com/multiversx/mx-chain-go/testscommon/bootstrapMocks"
+	cache2 "github.com/multiversx/mx-chain-go/testscommon/cache"
```
Comment (sstanculeanu) [inline integrationTests/testProcessorNode.go:111]: replace alias

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 2]
```diff
@@ -394,30 +400,58 @@ func (ihnc *indexHashedNodesCoordinator) ComputeConsensusGroup(
...
+}
+
+func (ihnc *indexHashedNodesCoordinator) selectLeaderAndConsensusGroup(
+	selector RandomSelector,
+	randomness []byte,
+	eligibleList []Validator,
+	consensusSize int,
+	epoch uint32,
+) (Validator, []Validator, error) {
+	if !ihnc.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, epoch) {
```
Comment (sstanculeanu) [inline sharding/nodesCoordinator/indexHashedNodesCoordinator.go:429]:
```suggestion
	if !ihnc.enableEpochsHandler.IsFlagEnabledInEpoch(common.FixedOrderInConsensusFlag, epoch) {
```
?

File: testscommon/shardedDataCacheNotifierMock.go [hunk 3]
```diff
@@ -4,7 +4,9 @@ import (
 	"sync"
 
 	"github.com/multiversx/mx-chain-core-go/core/counting"
+
 	"github.com/multiversx/mx-chain-go/storage"
+	cache2 "github.com/multiversx/mx-chain-go/testscommon/cache"
```
Comment (sstanculeanu) [inline testscommon/shardedDataCacheNotifierMock.go:9]: replace alias

File: sharding/nodesCoordinator/indexHashedNodesCoordinator_test.go [hunk 4]
```diff
@@ -2972,6 +2973,178 @@ func TestNodesCoordinator_CustomConsensusGroupSize(t *testing.T) {
...
+		arguments.EnableEpochsHandler = &enableEpochsHandlerMock.EnableEpochsHandlerStub{
+			IsFlagEnabledInEpochCalled: func(flag core.EnableEpochFlag, epoch uint32) bool {
+				return true
+			},
+		}
+
+		ihnc, err := NewIndexHashedNodesCoordinator(arguments)
+		require.Nil(t, err)
+
+		selector := &nodesCoordinatorMocks.RandomSelectorMock{
```
Comment (ssd04) [inline sharding/nodesCoordinator/indexHashedNodesCoordinator_test.go:3115]: add unit tests for fixed ordering in selectors implementations? or add unit tests here with real selector components

File: integrationTests/testProcessorNode.go (hunk 1 above)
Comment (AdoAdoAdo) [reply integrationTests/testProcessorNode.go:111]: done

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go (hunk 2 above)
Comment (AdoAdoAdo) [reply sharding/nodesCoordinator/indexHashedNodesCoordinator.go:429]: yes, replaced with the fixedOrder flag

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 5]
```diff
@@ -394,30 +400,58 @@ func (ihnc *indexHashedNodesCoordinator) ComputeConsensusGroup(
 		"round", round,
 		"shardID", shardID)
 
-	tempList, err := selectValidators(selector, randomness, uint32(consensusSize), eligibleList)
+	l, consensusGroup, err := ihnc.selectLeaderAndConsensusGroup(selector, randomness, eligibleList, consensusSize, epoch)
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply sharding/nodesCoordinator/indexHashedNodesCoordinator.go:403]: done

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 6]
```diff
@@ -40,6 +40,12 @@ type validatorWithShardID struct {
 	shardID   uint32
 }
 
+// SavedConsensusGroup holds the leader and consensus group for a specific selection
+type SavedConsensusGroup struct {
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply sharding/nodesCoordinator/indexHashedNodesCoordinator.go:44]: done

File: sharding/nodesCoordinator/indexHashedNodesCoordinator.go [hunk 7]
```diff
@@ -394,30 +400,58 @@ func (ihnc *indexHashedNodesCoordinator) ComputeConsensusGroup(
...
+		if len(tempList) == 0 {
+			return nil, nil, ErrEmptyValidatorsList
+		}
 
-	ihnc.consensusGroupCacher.Put(key, tempList, size)
+		return tempList[0], tempList, nil
+	}
 
-	return tempList[0], tempList, nil
+	selectedValidators, err := selectValidators(selector, randomness, 1, eligibleList)
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply sharding/nodesCoordinator/indexHashedNodesCoordinator.go:442]: done

**Review by sstanculeanu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**